### PR TITLE
Add --no-dependencies option to dotnet test (fixes #50640)

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Adds the `--no-dependencies` option to `dotnet test`, mirroring the existing option on `dotnet build`. The option forwards to MSBuild as `BuildProjectReferences=false`, so project-to-project references are skipped when building the test project.

Closes #50640.

## Fix

- `TestCommandDefinition.MicrosoftTestingPlatform.cs`: declare `NoDependenciesOption` and add to `Options`, forwarding `--property:BuildProjectReferences=false`.
- `TestCommandDefinition.VSTest.cs`: same option, forwarding `-property:BuildProjectReferences=false` (matching the existing VSTest property-prefix convention used elsewhere in that file).
- Snapshots updated: MTP help text, plus bash/pwsh/zsh completion snapshots.
- Unit tests added in `TestCommandParserTests.cs` verifying both runners forward the option correctly.

## Design notes

`dotnet build --no-dependencies` and `dotnet restore --no-dependencies` both already exist:

- `dotnet build --no-dependencies` forwards `--property:BuildProjectReferences=false` (skips building P2P refs).
- `dotnet restore --no-dependencies` forwards `-property:RestoreRecursive=false` (skips restoring P2P refs).

For `dotnet test`, the natural mapping is the build flavor: when the user opts out of dependencies, they want to skip *building* project-to-project references during the test run. The restore flavor is implicit anyway when `--no-restore`/`--no-build` are used. Forwarding `BuildProjectReferences=false` is consistent with `dotnet build --no-dependencies` and reuses the existing MSBuild property contract — no new SDK plumbing required.

## Alternatives considered

1. **Combined flavor (`BuildProjectReferences=false` + `RestoreRecursive=false`).** Rejected: `dotnet test` already invokes restore by default the same way `dotnet build` does. Mirroring `dotnet build --no-dependencies` keeps the semantics predictable. Users who want to also skip dependency restore can combine with `--no-restore`.
2. **MTP-only or VSTest-only.** Rejected: both runners go through MSBuild for the build step, so the option is meaningful in both contexts and should behave identically.
3. **Custom property name (e.g., `TestSkipDependencies`).** Rejected: introduces SDK surface area for no benefit, and tests are already built by `dotnet build` semantics — reusing `BuildProjectReferences=false` keeps the build behavior identical to `dotnet build --no-dependencies`.

## Tests

- `VSTestCommandIncludesNoDependenciesOption` — verifies VSTest forwards `-property:BuildProjectReferences=false`.
- `MTPCommandIncludesNoDependenciesOption` — verifies MTP forwards `--property:BuildProjectReferences=false`.

Both pass locally.

### CODE REVIEW

**Review by JanKrivanek — APPROVED** [review]
LGTM!

**Review by Evangelink — COMMENTED** [review]
## Expert Review — PR #54435: Add `--no-dependencies` option to `dotnet test`

> Automated expert review adapted from the dotnet/testfx expert-reviewer agent definition.

### Dimension Assessment

| # | Dimension | Status | Note |
|---|-----------|--------|------|
| 1 | Correctness | ✅ PASS | Forwards `BuildProjectReferences=false` to MSBuild correctly via `OptionValuesToBeForwarded` pipeline |
| 2 | Error Handling | ✅ PASS | N/A — boolean flag with no runtime error surface |
| 3 | Thread Safety | ✅ PASS | N/A — option declaration only |
| 4 | Performance | ✅ PASS | No runtime perf impact |
| 5 | API Design | ✅ PASS | Matches existing `dotnet build --no-dependencies` naming and semantics |
| 6 | Breaking Changes | ✅ PASS | Additive only — new option, no existing behavior changes |
| 7 | Security | ✅ PASS | N/A |
| 8 | Test Coverage | ✅ PASS | Tests verify forwarding strings for both MTP and VSTest paths |
| 9 | Code Style | ✅ PASS | Matches conventions in both definition files |
| 10 | Documentation | ✅ PASS | Help text and completions updated |
| 11 | Localization | ✅ PASS | Reuses existing `NoDependenciesOptionDescription` resource key from `CommandDefinitionStrings.resx`; no `.xlf` manual edits |
| 12 | Accessibility | ✅ PASS | N/A |
| 13 | Diagnostics | ✅ PASS | N/A |
| 14 | Compatibility | ✅ PASS | Consistent with `dotnet build`, `dotnet publish`, `dotnet pack` |
| 15 | Configuration | ✅ PASS | N/A |
| 16 | Resource Mgmt | ✅ PASS | N/A |
| 17 | Extensibility | ✅ PASS | N/A |
| 18 | Dependencies | ✅ PASS | Only adds necessary `using Microsoft.DotNet.Cli.CommandLine` for `ForwardAs` in MTP file |
| 19 | Architecture | ✅ PASS | Minimal, single-concern change |
| 20 | Edge Cases | ⚠️ CONCERN | See SUGGESTION below regarding `--no-build` + `--no-dependencies` interaction |
| 21 | Snapshot Tests | ✅ PASS | All three shell completions + MTP help snapshot updated correctly |

### Findings

#### SUGGESTION 1: Interaction with `--no-build`

When `--no-build` is set, MTP skips the build phase entirely (`MSBuildUtility.cs:235`), making `--no-dependencies` a silent no-op. This is consistent with how `dotnet build --no-build --no-dependencies` behaves (the flag is simply not meaningful), but consider adding a note in the option description or docs clarifying this, to avoid user confusion.

**Severity**: SUGGESTION — no code change required.

#### SUGGESTION 2: Consider consolidating as a shared factory method

`BuildCommandDefinition`, `PublishCommandDefinition`, `PackCommandDefinition`, and `RestoreCommandDefinition` all declare their own local `NoDependenciesOption` field with the same pattern. A shared `CommonOptions.CreateNoDependenciesOption()` factory (like `CreateNoRestoreOption()`) could reduce duplication. However, this is a pre-existing pattern and the PR correctly follows it — so no action needed here.

**Severity**: SUGGESTION — future cleanup opportunity, not this PR's concern.

#### SUGGESTION 3: Integration test coverage

The unit tests verify the option is forwarded as the correct MSBuild property. For full confidence, an integration test that invokes `dotnet test --no-dependencies` against a project with P2P refs and verifies that dependencies are NOT built would add value. However, since the mechanism is purely MSBuild property passthrough (identical to `dotnet build --no-dependencies`), the unit-level forwarding tests are sufficient for this PR.

**Severity**: SUGGESTION — nice-to-have, not blocking.

### Summary Verdict

**✅ APPROVE-worthy** — Clean, minimal, well-tested change that correctly adds the `--no-dependencies` option to both MTP and VSTest code paths. Property prefix conventions are correct (`--property:` for MTP matching `BuildCommandDefinition`, `-property:` for VSTest matching its existing options). Localization is handled via existing resource key. No blocking or major issues found.

---
*This review was produced by an automated expert reviewer adapted from the dotnet/testfx expert-reviewer agent definition.*

File: src/Cli/Microsoft.DotNet.Cli.Definitions/Commands/Test/TestCommandDefinition.VSTest.cs [hunk 1]
```diff
@@ -71,6 +71,12 @@ public sealed class VSTest : TestCommandDefinition
             Arity = ArgumentArity.Zero
         }.ForwardAs("-property:VSTestNoBuild=true");
 
+        public readonly Option<bool> NoDependenciesOption = new Option<bool>("--no-dependencies")
+        {
+            Description = CommandDefinitionStrings.NoDependenciesOptionDescription,
+            Arity = ArgumentArity.Zero
+        }.ForwardAs("-property:BuildProjectReferences=false");
```
Comment (Youssef1313) [inline src/Cli/Microsoft.DotNet.Cli.Definitions/Commands/Test/TestCommandDefinition.VSTest.cs:78]: Is it intended to add this to VSTest?

Discussion in original issue was a "no for VSTest"
